src/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -2153,9 +2154,13 @@ protected boolean isGenericDescriptor(ScriptableObject desc) {
     }
 
     protected static Scriptable ensureScriptable(Object arg) {
-        if ( !(arg instanceof Scriptable) )
-            throw ScriptRuntime.typeError1("msg.arg.not.object", ScriptRuntime.typeof(arg));
-        return (Scriptable) arg;
+        if (arg instanceof Scriptable) {
+            return (Scriptable) arg;
+        }
+        if (arg instanceof Delegator) {
```

Comment:
OK -- this makes more sense now. Still, Delegator is a subclass of Scriptable so this particular test will never be "true." If the intention is to get the "delegee" then do we want to have this test first?